UM2N/generator/unstructured_mesh.py

```diff
@@ -0,0 +1,194 @@
+    """
+    Create a random polygonal mesh by spliting the edge of a
+    square randomly.
+    """
+
+    def __init__(self, scale=1.0, mesh_type=2):
+        """
+        :kwarg scale: overall scale factor for the domain size
+        :type scale: float
+        :kwarg mesh_type: Gmsh algorithm number
+        :type mesh_type: int
+        """
+        super().__init__(mesh_type=mesh_type)
+        self.scale = scale
+
+    @staticmethod
+    def sample_uniform(mean, interval):
+        """
+        Sample a point from a uniform distribution with a given mean and interval.
+
+        Note that the interval is *either side* of the mean, not the overall interval.
+
+        :arg mean: the mean value of the uniform distribution
+        :type mean: float
+        :arg interval: the interval either side of the mean
+        :type interval: float
+        :returns: sampled point
+        :rtype: float
+        """
+        return random.uniform(mean - interval, mean + interval)
+
+    @property
+    def corners(self):
+        """
+        :returns: corner vertices of a randomly generated polygonal domain
+        :rtype: tuple
+        """
+        if hasattr(self, "_corners"):
+            return self._corners
+        start = 0
+        finish = self.scale
+        split_threshold = 0.3
+        mid = (start + finish) / 2
+        quarter = (start + mid) / 2
+        three_quarter = (mid + finish) / 2
+        mid_interval = (finish - start) / 3
+        quarter_interval = (mid - start) / 4
+        points = []
+        split_p = np.random.uniform(0, 1, 4)
+        # edge 1
+        if split_p[0] < split_threshold:
+            points.append([self.sample_uniform(quarter, quarter_interval), 0])
+            points.append([self.sample_uniform(three_quarter, quarter_interval), 0])
+        else:
+            points.append([self.sample_uniform(mid, mid_interval), 0])
+        # edge 2
+        if split_p[1] < split_threshold:
+            points.append([self.scale, self.sample_uniform(quarter, quarter_interval)])
+            points.append(
+                [self.scale, self.sample_uniform(three_quarter, quarter_interval)]
+            )
+        else:
+            points.append([self.scale, self.sample_uniform(mid, mid_interval)])
+        # edge 3
+        if split_p[2] < split_threshold:
+            points.append(
+                [self.sample_uniform(three_quarter, quarter_interval), self.scale]
+            )
+            points.append([self.sample_uniform(quarter, quarter_interval), self.scale])
+        else:
+            points.append([self.sample_uniform(mid, mid_interval), self.scale])
+        # edge 4
+        if split_p[3] < split_threshold:
+            points.append([0, self.sample_uniform(three_quarter, quarter_interval)])
+            points.append([0, self.sample_uniform(quarter, quarter_interval)])
+        else:
+            points.append([0, self.sample_uniform(mid, mid_interval)])
```

Comment:
No need to review this, it was more or less copied over from the previous implementation.